### ISSUE TITLE
`dealiasing` should be part of the `equation`, not the `grid`

### ISSUE THREAD

**Issue by navidcy** [FourierFlows/FourierFlows.jl#286]
It makes more sense.

Yes, dealiasing is related to wavenumbers. But whether dealiasing should happened and what sort of dealiasing (2/3, 1/2, ...) depends on the order of nonlinearity that `calcN!` includes. So it makes more sense to make `kralias`, `kalias`, ...  part of the `Equation` struct.

**Comment by glwagner** [comment]
I think this could work.

With a fully-fledged field abstraction, I think dealiasing would have to be part of the grid. The reason is that dealiasing as a concept is really associated with arithmetic operations between fields that have a discrete spectral representation (eg multiplying two fields together), rather than being associated with a PDE.

**Comment by navidcy** [comment]
Hm... OK. Dealiasing should be part of the grid. But for different equations the `aliasing_factor` could be different...

**Comment by navidcy** [comment]
We'll leave this for future, not something to be done in #285.

**Comment by glwagner** [comment]
I guess equations / problems might have to enforce certain grid properties?

**Comment by navidcy** [comment]
Yes, ideally either the equation or the problem might be able to change the `aliased_fraction` property of the grid...
